test/unit/nested.cpp

```diff
@@ -0,0 +1,31 @@
+#include "gtest/gtest.h"
+
+#include "RAJA/pattern/nested.hpp"
+
+#include <iostream>
+
+TEST(Nested, Basic)
+{
+  using namespace RAJA::nested;
+  using Index_type = RAJA::Index_type;
+  using pol = Policy<For<1, RAJA::seq_exec>, For<0, RAJA::seq_exec>>;
+  RAJA::nested::forall(pol{},
+                       RAJA::util::make_tuple(RAJA::RangeSegment(0, 5),
+                                              RAJA::RangeSegment(0, 5)),
+                       [=](Index_type i, Index_type j) {
+                         std::cout << i << " " << j << std::endl;
+                       });
+}
+
+TEST(Nested, collapse)
+{
+  using namespace RAJA::nested;
+  using Index_type = RAJA::Index_type;
+  using pol = Policy<Collapse<RAJA::seq_exec, CFor<1>, CFor<0>>>;
```

Comment:
@willkill07 what are you talking about? the old forallN policies were trivial to work with :)
jk... I'm glad they are clearer to you.

Also, this will make it possible to use ordinary RAJA exec polices on the tiling loops, something @tepperly has been asking for.